full_storage_utilization_test.py

```diff
@@ -0,0 +1,195 @@
+        self.run_stress_until_target(target_used_size, target_usage)
+
+        self.log_disk_usage()
+        self.log.info(f"Wait for {sleep_time} seconds")
+        time.sleep(sleep_time)  
+        self.log_disk_usage()
+
+    def run_stress_until_target(self, target_used_size, target_usage):
+        current_used = self.get_max_disk_used()
+        current_usage = self.get_max_disk_usage()
+        num = 0
+        smaller_dataset = False
+        
+        space_needed = target_used_size - current_used
+        # Calculate chunk size as 10% of space needed
+        chunk_size = int(space_needed * 0.1)
+        while current_used < target_used_size and current_usage < target_usage:
+            num += 1
+            
+            # Write smaller dataset near the threshold (15% or 30GB of the target)
+            smaller_dataset = (((target_used_size - current_used) < 30) or ((target_usage - current_usage) <= 15))
+
+            # Use 1GB chunks near threshold, otherwise use 10% of remaining space
+            dataset_size = 1 if smaller_dataset else chunk_size
+            ks_name = "keyspace_small" if smaller_dataset else "keyspace_large"
+            self.log.info(f"Writing chunk of size: {dataset_size} GB")
+            stress_cmd = self.prepare_dataset_layout(dataset_size)
+            stress_queue = self.run_stress_thread(stress_cmd=stress_cmd, keyspace_name=f"{ks_name}{num}", stress_num=1, keyspace_num=num)
+
+            self.verify_stress_thread(cs_thread_pool=stress_queue)
+            self.get_stress_results(queue=stress_queue)
+
+            self.flush_all_nodes()
+            #time.sleep(60) if smaller_dataset else time.sleep(600)
+
+            current_used = self.get_max_disk_used()
+            current_usage = self.get_max_disk_usage()
+            self.log.info(f"Current max disk usage after writing to keyspace{num}: {current_usage}% ({current_used} GB / {target_used_size} GB)")
+
+    def add_new_node(self):
+        new_nodes = self.db_cluster.add_nodes(count=self.add_node_cnt, enable_auto_bootstrap=True)
+        self.db_cluster.wait_for_init(node_list=new_nodes)
+        self.db_cluster.wait_for_nodes_up_and_normal(nodes=new_nodes)
+        total_nodes_in_cluster = len(self.db_cluster.nodes)
+        self.log.info(f"New node added, total nodes in cluster: {total_nodes_in_cluster}")
+        self.monitors.reconfigure_scylla_monitoring()
+        
+    def remove_node(self):  
+        self.log.info('Removing a first node from the cluster')
+        node_to_remove = self.db_cluster.nodes[0]
+        self.log.info(f"Node to be removed: {node_to_remove.name}")
+        self.db_cluster.decommission(node_to_remove)
+        self.db_cluster.wait_for_nodes_up_and_normal(nodes=[node_to_remove])
+        self.log.info(f"Node {node_to_remove.name} has been removed from the cluster")
+        self.monitors.reconfigure_scylla_monitoring()
+    
+    def flush_all_nodes(self):
```

Comment:
the below util methods possibly already exist under sstabe-util or other utils directory.

Reply:
grep -r 'node.run_nodetool("flush")' .` or  `gep -r 'flush_all'`  shows no util methods. longevity_test.py has similar function but not from util should we create one and add to util and update this file along with longevity_test.py?

Reply:
there's `_flush_all_nodes()` function which sits on longevity_test.py, and not used at all.
there are more places in the code, in manager test doing flush on all nodes.

this kind of a utility function, can be put on `BaseScyllaCluster` for example, so it can be used from multiple places

Reply:
ok, added into BaseScyllaCluster and using it from there.